### PR TITLE
[EMCAL-613] Converting digitizer and digits writer spec to DPL task

### PR DESCRIPTION
**Pull request by mfasDa**
- Lambda functions coverted to init and run function
- Local functions to large extend class members of the task
- Documentation added

### CODE REVIEW

**Comment by mfasDa** [comment]
@sawenzel Please review!

File: Steer/DigitizerWorkflow/src/EMCALDigitWriterSpec.cxx [hunk 1]
```diff
@@ -41,72 +36,63 @@ TBranch* getOrMakeBranch(TTree& tree, std::string brname, T* ptr)
   return tree.Branch(brname.c_str(), ptr);
 }
 
+void DigitsWriterSpec::init(framework::InitContext& ctx)
+{
```
Comment (sawenzel) [inline Steer/DigitizerWorkflow/src/EMCALDigitWriterSpec.cxx]: Wondering if you could not use the generic ROOTWriterSpec (as used by some other digit writers)?
Would make the code shorter.

Comment (mfasDa) [reply Steer/DigitizerWorkflow/src/EMCALDigitWriterSpec.cxx]: @sawenzel For the raw data encoding we will need to perform a couple of changes in order to account for the triggered mode, particularly moving the digits time from absolute time to time - trigger, which means that we need to outsource the trigger time (time of the event where we open the readout) and the bunch crossing index to a header which we add to the tree as separate branch - not sure this is supported by the ROOTWriterSpec (@matthiasrichter ).

**Comment by mfasDa** [comment]
@sawenzel Can this be merged?

**Pull request merged**